js/pages/settings.js

```diff
@@ -173,44 +191,45 @@ const handleEditNameSection = () => {
       middleName = middleNameField.value.trim();
       isNameFormDisplayed = toggleElementVisibility(nameElementArray, isNameFormDisplayed);
       toggleButtonText();
-
-      changeName(firstName, lastName, middleName, suffix, preferredFirstName)
-        .then(() => {
-          handleOptionalFieldVisibility(middleName, 'profileMiddleName', middleNameRow, nameElementArray[0], 'text');
-          handleOptionalFieldVisibility(suffix, 'profileSuffix', suffixRow, nameElementArray[0], 'suffix');
-          handleOptionalFieldVisibility(preferredFirstName, 'profilePreferredFirstName', preferredFirstNameRow, nameElementArray[0], 'text');
-          successMessageElement = document.getElementById('changeNameSuccess');
-          successMessageElement.style.display = 'block';
-          document.getElementById('profileFirstName').textContent = firstName;
-          document.getElementById('profileLastName').textContent = lastName;
-        })
-        .catch(function (error) {
-          document.getElementById('changeNameFail').style.display = 'block';
-          document.getElementById('changeNameError').innerHTML = error.message;
-        });
+      submitNewName(firstName, lastName);
     }
   });
 };
 
+const submitNewName = async (firstName, lastName) => {
+  const isSuccess = await changeName(firstName, lastName, middleName, suffix, preferredFirstName, userData).catch(function (error) {
```

Comment:
are these global variables? where are parameters like `middleName` coming from?

Reply:
yes - same as the above - the nullable fields are declared as globals so behavior can be responsive with the way the user edits the form.